### PR TITLE
[autoparallel] Patch meta information of `torch.nn.Embedding`

### LINKED ISSUE #2735: [FEATURE]: Patch meta information of `torch.nn.Embedding`
This is a part of issue #2628, we will patch meta information of `torch.nn.Embedding`

### PR DESCRIPTION
**Pull request by Cypher30**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`
Resolved #2735 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

In this PR, I patch meta information of `torch.nn.Embedding`, and fix some small errors caused by several former PRs related to meta information patch.

***NOTE***: the temp memory during the backward phase of `torch.nn.Embedding` is quite weird (sometimes it doesn't have temp memory) and currently I just simply set it to zero, as in NLP tasks the temp memory cost is significantly smaller than gradient memory cost.

The tests will be skipped when running on torch 1.11.0, so I attach the test results here
Screenshot 2023-02-16 at 17 12 30

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: tests/test_auto_parallel/test_tensor_shard/test_metainfo/test_activation_metainfo.py [hunk 1]
```diff
@@ -104,5 +104,5 @@ def test_sofmax_meta_info():
 
 
 if __name__ == '__main__':
-    # test_ReLU_meta_concrete_info_match()
+    test_ReLU_meta_concrete_info_match()
```
Comment (YuliangLiu0306) [inline tests/test_auto_parallel/test_tensor_shard/test_metainfo/test_activation_metainfo.py:107]: Please ename this function to test_relu_meta_concrete_info_match

Comment (Cypher30) [reply tests/test_auto_parallel/test_tensor_shard/test_metainfo/test_activation_metainfo.py]: I decide to remove those changes and solve them in the next PR.

**Review by YuliangLiu0306 — APPROVED** [review]
